ocaml/xapi/xapi.ml

```diff
@@ -516,6 +516,21 @@ let server_run_in_emergency_mode () =
   wait_to_die () ; exit 0
 
 let update_certificates ~__context () =
+  let module GencertService = Helpers.SystemdService (struct
```

Comment:
Is this not a problem that should be solved by service dependencies?

Reply:
I've asked Ben the same question elsewhere, it can't be done unless a third service sets MANAGEMENT_INTERFACE in xensource-inventory. 

> I'm wondering if the dependency could be done with the systemd unit files instead of a waiting loop in xapi, was there something that made you avoid that solution?

> that won't work (at the moment, maybe there is a better way) because xapi sets MANAGEMENT_INTERFACE in xensource-inventory, which gencert needs